src/malli/design_draft.clj

```diff
@@ -0,0 +1,96 @@
+(ns malli.design-draft
+  (:import (clojure.lang IPersistentMap)))
+
+
+;;; malli schemas
+
+;; Core schemas are not namespace qualified keywords to make using them user
+;; friendly.
+
+;; Schemas are written using Hiccup style vectors, where order is significant.
+
+;; Every schema definition can have additional options-map after schema name.
+;; Options-map can have arbitrary amount of data, with some keys having special
+;; meaning, like below, which describes a closed map.
+
+[:map {:closed true, :description "My closed map"}]
+
+;; Schema definitions live in registry. Malli provides a core registry for
+;; all the essential schemas. Registry can be extended on the runtime and it's
+;; possible to have as many registries one needs.
+
+;; The actual registry definitions and format is wip, but could be something
+;; like below.
+
+;; Compiles given definition to Schema
+(declare compile)
+(declare Schema)
+
+(def -map (compile {:id :map
```

Comment:
agree with @nilern that direct impls for core things (like there is in the core now) is clearer. The core should be small and separate concerns, like emitting json-schemas, should be in separate namespaces. At bare minimum, the core only has models and validation. Coercion and generation and definitely model transformations, should be optional.

for end-uses, support of extension via map data is a good idea. e.g. define a single `my-uuid` and attach all the possible things to it via namespaced keys in the schema data.

Reply:
Ok, sounds good, I'll try to reflect that on the update to draft.